app/policies/partner_policy.rb

```diff
@@ -11,11 +11,11 @@ def create?
   end
 
   def show?
-    record == user
+    user&.partner == record
   end
 
   def update?
-    record == user
+    user&.partner == record
```

Comment:
wondering about the lonely operators ^^ and if they're needed. i would expect users to always have a partner except if they're an admin.

Reply:
Users will always have an admin, but we test unauthenticated requests in a couple of places that necessitates this. Otherwise instead of denying access we throw and error.